### DISCUSSION TITLE
register_fprime_module() and header files

### DISCUSSION THREAD

**Discussion by Gonta01** [nasa/fprime#2824] [Q&A] [answered]
As specified in [Cmake API](https://nasa.github.io/fprime/UsersGuide/api/cmake/API.html) when we use register_fprime_module() we need to specify two variables SOURCE_FILES and MOD_DEPS. 
Could somebody explain, how header files that are near specified SOURCE_FILES are made visible to them.

**Comment by LeStarch** [comment, marked as answer]
Headers are located from the following directories:

1. Root of project
2. Root of any "fprime libraries"
3. Root of fprime framework
4. Root of project in build cache (for autocoded files)
5. Root of fprime libraries in build cache (for autocoded files)
6. Root of fprime framework in build cache (for autocoded files)

If you need special paths you can use `target_include_directories` or `include_directories` CMake commands to add them
